ps_facebook.php

```diff
@@ -167,12 +167,15 @@ public function __construct()
         $this->docs_path = $this->_path . 'docs/';
         $this->confirmUninstall = $this->l('Are you sure you want to uninstall this module?');
         $this->ps_versions_compliancy = ['min' => '1.7.0.0', 'max' => _PS_VERSION_];
-        $this->front_controller = $this->context->link->getModuleLink(
-            $this->name,
-            'FrontAjaxPixel',
-            [],
-            true
-        );
+
+        if ($this->context->link !== null) {
```

Comment:
Link is supposed to be always defined, let's add a comment to avoid reverting this change later

```suggestion
        // Needed PrestaShop installation, where Link class in not initialized 
        if ($this->context->link !== null) {
```